test/core/event_engine/work_queue/work_queue_fuzzer.cc

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <grpc/support/port_platform.h>
+
+#include <cassert>
+#include <deque>
+
+#include "absl/container/flat_hash_map.h"
+
+#include "src/core/lib/event_engine/common_closures.h"
+#include "src/core/lib/event_engine/work_queue.h"
+#include "src/libfuzzer/libfuzzer_macro.h"
+#include "test/core/event_engine/work_queue/work_queue_fuzzer.pb.h"
+
+bool squelch = true;
+bool leak_check = true;
+
+namespace grpc_event_engine {
+namespace experimental {
+
+class WorkQueueFuzzer {
+ public:
+  WorkQueueFuzzer() { CheckEqual(); };
+  ~WorkQueueFuzzer() { CheckEqual(); };
+
+  void Run(const work_queue_fuzzer::Action& action) {
+    switch (action.action_type_case()) {
+      case work_queue_fuzzer::Action::kAdd: {
+        if (action.add().type() == work_queue_fuzzer::CALLBACK_TYPE_CLOSURE) {
+          work_queue_.Add(CreateClosure(action.add().key()));
+          deque_.push_back(CreateClosure(action.add().key()));
+        } else {
+          work_queue_.Add(CreateInvocable(action.add().key()));
+          deque_.push_back(CreateClosureWrappedInvocable(action.add().key()));
+        }
+      } break;
+      case work_queue_fuzzer::Action::kPopFront: {
+        // pop and discard front closure, executing to ensure memory freeing
+        auto* wq_c = work_queue_.PopFront();
+        if (wq_c == nullptr) {
+          assert(deque_.empty());
```

Comment:
use GPR_ASSERT or something that doesn't disappear in opt builds (since we'll also fuzz in opt builds)

Reply:
or do the abort like on 73

Reply:
I replaced both instances with a variation of the abort.